src/pages/index.js

```diff
@@ -197,32 +197,21 @@ function handleNewPlaceSubmit(placeCardData) {
           link: newPlaceCard.link,
           _id: newPlaceCard._id,
         });
+        addPlaceModal.reset();
         section.addItem(cardElement);
       });
   }
-
-  handleSubmit(makeRequest, addPlaceModal, "Saving...").then(() => {
-    addPlaceModal.reset();
+  handleSubmit(makeRequest, addPlaceModal, "Saving...", "Save", () => {
     addPlaceModal.setLoading(true, "Save");
```

Comment:
You don't need this callback here in `handleSubmit` because you have everything inside `handleSubmit`


```js
  handleSubmit(makeRequest, addPlaceModal, "Saving...", "Save");


```